src/components/textarea/textarea.component.js

```diff
@@ -20,196 +20,180 @@ const getFormatNumber = (value, locale) =>
 const marginPropTypes = filterStyledSystemMarginProps(
   styledSystemPropTypes.space
 );
-class Textarea extends React.Component {
-  // Minimum height of the textarea
-  minHeight = 0;
 
-  id = this.props.id || guid();
+const Textarea = ({
+  autoFocus,
+  fieldHelp,
+  label,
+  size,
+  children,
+  characterLimit,
+  enforceCharacterLimit,
+  warnOverLimit,
+  onChange,
+  disabled,
+  labelInline,
+  labelAlign,
+  labelHelp,
+  labelSpacing,
+  inputIcon,
+  id = guid(),
+  error,
+  warning,
+  info,
+  name,
+  readOnly,
+  placeholder,
+  expandable,
+  rows,
+  cols,
+  validationOnLabel,
+  adaptiveLabelBreakpoint,
+  inputWidth,
+  labelWidth,
+  tooltipPosition,
+  value,
+  "data-component": dataComponent,
+  "data-element": dataElement,
+  "data-role": dataRole,
+  ...props
+}) => {
+  const locale = useContext(LocaleContext);
 
-  _input = React.createRef();
+  const inputRef = useRef();
 
-  /**
-   * A lifecycle method that is called after initial render.
-   * Allows access to refs and DOM to set expandable variables
-   */
-  componentDidMount() {
-    if (this.props.expandable) {
-      window.addEventListener("resize", this.expandTextarea);
-      // Set the min height to the initially rendered height.
-      // Without minHeight expandable textareas will only have
-      // one line when no content is present.
-      this.minHeight = this._input.current.clientHeight;
-
-      this.expandTextarea();
-    }
-  }
-
-  componentWillUnmount() {
-    if (this.props.expandable) {
-      window.removeEventListener("resize", this.expandTextarea);
-    }
-  }
-
-  componentDidUpdate() {
-    const { expandable } = this.props;
-
-    if (expandable) {
-      this.expandTextarea();
-    }
-  }
+  const minHeight = useRef(0);
 
-  expandTextarea = () => {
-    const textarea = this._input.current;
+  const expandTextarea = () => {
+    const textarea = inputRef.current;
 
-    if (textarea.scrollHeight > this.minHeight) {
+    if (textarea.scrollHeight > minHeight.current) {
       // Reset height to zero - IE specific
       textarea.style.height = "0px";
       // Set the height so all content is shown
       textarea.style.height = `${Math.max(
         textarea.scrollHeight,
-        this.minHeight
+        minHeight.current
       )}px`;
     }
   };
 
-  get overLimit() {
-    const value = this.props.value || "";
-    return value.length > parseInt(this.props.characterLimit, 10);
-  }
+  useEffect(() => {
+    if (expandable) {
+      expandTextarea();
+    }
+  });
```

Comment:
```suggestion
  }, [expandable]);
```

Reply:
This one has to be run on every component update, not only when `expandable` prop changes. Previously it was in `componentDidUpdate` lifecycle method

Reply:
Yeah good point I hadn't considered you'd not set an array at all